### DISCUSSION TITLE
Building docker image, lib not found

### DISCUSSION THREAD

**Discussion by kakkoiiAllen** [milvus-io/milvus#18550] [Q&A and General discussion]
Hi:
I'm building milvus from source. I run ./build/builder.sh make and I get a bin/milvus file. Now I'd like to build the docker image. I tried ./build/build_image.sh, but I got lib not found. Where can I get the lib directory? Can I just copy the lib dir from public docker image milvusdb/milvus? Thx.

OS: MacOS(Intel Chip)
Memory: 64GB

<img width="690" alt="Screen Shot 2022-08-06 at 7 49 45 AM" src="https://user-images.githubusercontent.com/94626600/183254843-2bd8b8ad-3bf5-41ad-b491-e1493e96f8ff.png">

update: I tried "make docker" and it fails too
<img width="698" alt="Screen Shot 2022-08-06 at 8 38 49 AM" src="https://user-images.githubusercontent.com/94626600/183256207-ae6764c4-3336-4b35-a239-70be50d50d5e.png">

update: I tried v2.0.2 and v2.1.0 version and both built successfully, but the docker image shows exec format error. I assume that after running "make docker", I can get an image and run right away using "docker run" command. 
<img width="572" alt="Screen Shot 2022-08-06 at 9 42 10 AM" src="https://user-images.githubusercontent.com/94626600/183258140-1bf3a350-6997-4d1f-90af-b8804f8fe66d.png">

To reproduce the problem:
cd milvus && git checkout v2.0.2 && make docker
docker run milvusdb/milvus:latest

**Comment by yhmo** [comment]
In the "Makefile" under the project, we can see:
```
LIBRARY_PATH := $(PWD)/lib
# Build each component and install binary to $GOPATH/bin.
install: all
	@echo "Installing binary to './bin'"
	@mkdir -p $(GOPATH)/bin && cp -f $(PWD)/bin/milvus $(GOPATH)/bin/milvus
	@mkdir -p $(LIBRARY_PATH) && cp -r -P $(PWD)/internal/core/output/lib/*.so* $(LIBRARY_PATH)
	@echo "Installation successful."
```

So, you can use "make install" command to copy necessary libraries into lib folder.

**Comment by LoveEachDay** [comment]
@kakkoiiAllen  Since you build the binary in mac, the binary is for mac, not for Linux. You'd better run the following command to build linux binary in mac:
```
cd milvus 
./build/builder.sh make install # build binary
docker build -f build/docker/milvus/ubuntu18.04/Dockerfile -t tag .  # build a docker image
```

**Reply by kakkoiiAllen** [reply]
@LoveEachDay Thank you so much! While building binary with "./build/builder.sh make install", I got error message requiring CUDA 8.0 and cpp build failed. Should I have CUDA toolkit to build the image? 
<img width="640" alt="Screen Shot 2022-08-08 at 8 12 34 AM" src="https://user-images.githubusercontent.com/94626600/183451918-70230ac5-077f-4c67-a24c-b46ac52ddbae.png">
<img width="1343" alt="Screen Shot 2022-08-08 at 8 39 34 AM" src="https://user-images.githubusercontent.com/94626600/183457542-4af1b1f7-c254-459d-a9df-c5c0d332855a.png">
